lib/nexpose/ajax.rb

```diff
@@ -158,14 +158,28 @@ def request(nsc, request, timeout = nil)
           raise Nexpose::AuthenticationFailed.new(response)
         else
           req_type = request.class.name.split('::').last.upcase
-          raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{request.body}", response.code)
+          raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{get_reason(request, response)}", response.code)
         end
       else
         req_type = request.class.name.split('::').last.upcase
-        raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{request.body}", response.code)
+        raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{get_reason(request, response)}", response.code)
       end
     end
 
+    def get_api_version(request)
+      begin
```

Comment:
Using the `begin` statement isn't required at the top level of a method (`def`/`class`/`module`) can all accept `rescue` implicitly.